lib/common/domain/check_if_app_terms_should_be_shown_usecase.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:zachranobed/services/auth_service.dart';
+import 'package:zachranobed/features/appConfiguration/data/get_last_app_terms_version_use_case.dart';
+
+class CheckIfAppTermsShouldBeShownUseCase {
+  final AuthService authService;
+  final GetLastAppTermsVersionUseCase _getLastAppTermsVersionUseCase;
+
+  CheckIfAppTermsShouldBeShownUseCase(
+      this.authService,
+      this._getLastAppTermsVersionUseCase
+  );
+
+  Future<bool> checkIfAppTermsShouldBeShown() async {
+    int? lastAppTermsVersion = await _getLastAppTermsVersionUseCase.getLastAppTerms();
+    var userData = authService.getUserData();
+
+    return userData.then((value) {
+      var lastAcceptedVersion = value?.lastAcceptedAppTermsVersion;
+
+      if (lastAcceptedVersion != null && lastAppTermsVersion != null) {
+        return lastAppTermsVersion > lastAcceptedVersion;
+      } else {
+        return true;
+      }
+    });
+  }
```

Comment:
Nit: Instead of `then` you could use `await` also on `authService.getUserData()` as you do with `getLastAppTerms`, so that the function is more "plain" (less nested)

Also I would use `final` instead of mutable `var` fields

```dart
Future<bool> checkIfAppTermsShouldBeShown() async {
  final lastAppTermsVersion = await _getLastAppTermsVersionUseCase.getLastAppTerms();
  final userData = await authService.getUserData();
  final lastAcceptedVersion = userData?.lastAcceptedAppTermsVersion;
  if (lastAcceptedVersion != null && lastAppTermsVersion != null) {
    return lastAppTermsVersion > lastAcceptedVersion;
  } else {
    return true;
  }
}
```

Reply:
Changed